simulation_parameters/Constants.cs

```diff
@@ -340,19 +345,57 @@ public static class Constants
     public const float ENGULFING_MOVEMENT_DIVISION = 1.7f;
 
     /// <summary>
-    ///   The speed reduction when a cell is being engulfed.
+    ///   The minimum size ratio between a cell and a possible engulfing victim.
     /// </summary>
-    public const float ENGULFED_MOVEMENT_DIVISION = 10.0f;
+    public const float ENGULF_SIZE_RATIO_REQ = 1.5f;
 
     /// <summary>
-    ///   The minimum size ratio between a cell and a possible engulfing victim.
+    ///   The duration for which an engulfable object can be engulfed again after being expelled.
     /// </summary>
-    public const float ENGULF_SIZE_RATIO_REQ = 1.5f;
+    public const float ENGULF_EJECTED_COOLDOWN = 2.0f;
+
+    public const float ENGULF_EJECTION_FORCE = 20f;
+
+    /// <summary>
+    ///   Offsets how far should the chunks for expelled partially digested objects be spawned from the membrane.
+    ///   0 means no offset and chunks are spawned directly on the membrane point.
+    /// </summary>
+    public const float EJECTED_PARTIALLY_DIGESTED_CELL_CORPSE_CHUNKS_SPAWN_OFFSET = 1.8f;
+
+    /// <summary>
+    ///   The measure of which beyond this threshold an engulfable is considered partially digested.
+    ///   Used to determine whether a cell should be able to heal after being expelled from engulfment.
+    /// </summary>
+    public const float PARTIALLY_DIGESTED_THRESHOLD = 0.5f;
 
     /// <summary>
-    ///   The amount of hp per second of damage when being engulfed
+    ///   The maximum digestion progress in which an engulfable is considered fully digested.
     /// </summary>
-    public const float ENGULF_DAMAGE = 45.0f;
+    public const float FULLY_DIGESTED_LIMIT = 1.0f;
+
+    /// <summary>
+    ///   The speed of which a cell can absorb compounds from digestible engulfed objects.
+    /// </summary>
+    public const float ENGULF_COMPOUND_ABSORBING_PER_SECOND = 0.5f;
+
+    /// <summary>
+    ///   How much compounds a cell can absorb per second from digestible engulfed objects.
+    /// </summary>
+    public const float ENGULF_BASE_COMPOUND_ABSORBTION_YIELD = 0.3f;
+
+    public const float ENGULF_TOXIC_COMPOUND_ABSORPTION_DAMAGE_FRACTION = 0.9f;
+
+    /// <summary>
+    ///   Each enzyme addition grants 20% increase in digestion speed.
```

Comment:
20 or 10% percent? the comment vs value seems to disagree, so probably reword the commend to not have the value in it, that way the comment doesn't need to be updated when the value is changed.